### DISCUSSION TITLE
add_blockref (problem)

### DISCUSSION THREAD

**Discussion by Quasarrs** [mozman/ezdxf#942] [Issues]
I encountered such a problem, I created a block in AUTOcade in which I placed an attribute from multi-line text. Next, I open this block in the code and try to reproduce it with copies using the method
self.msp.add_blockref("name_block", insert=(x_block, y_block)
I save, after opening, copies of this block were actually created. But for some reason the attribute in the mine became a one-line one.

How can I create copies of the block and at the same time preserve the multi-line nature of the built-in attribute?

**Comment by mozman** [comment]
Hi!

Multi-line attributes were introduced in DXF R2018 and are not documented. Avoid them if you can.

If you really have to/want to use multi-line attributes, you have to provide more context: test data in form of simplified and zipped DXF files and Python code to reproduce the issue.

**Reply by Quasarrs** [reply]
It turns out that if I now manually add blocks to the drawing file, inside which there are attributes with multi-line text. And then, using ezdxf, I want to read this file, take a block and save copies of it in MSP, then when opening the attributes that were multi-line will become single-line and nothing can be done about it?

**Comment by Quasarrs** [comment]
It's quite simple.
I created a block with a multi-line attribute in a CAD program, removed it from the model and saved it in DXF18.

Next is a simple program that takes a block from the file and adds it to the model as a copy.
It adds a block, but the block’s built-in attribute disappears ((((

```python
import ezdxf
dxf = ezdxf.readfile("data/test_m_text.dxf")
msp = dxf.modelspace()
block = msp.add_blockref("test_b", insert=(0, 0))
dxf.saveas("test_out_m_text.dxf")
``` 


In the photo there is a block that I manually took out from the block library and it has a multi-line attribute.
![image](https://github.com/mozman/ezdxf/assets/145274914/335a3b98-b67a-4d06-bbf0-c96eec5f48a0)
The copy made by the program has no attribute.
![image](https://github.com/mozman/ezdxf/assets/145274914/bc0828d1-8797-4f5a-87f4-2dce7d8d1d5b)
